crypto/src/zcash_format.rs

```diff
@@ -45,6 +78,32 @@ pub fn parse_hex(hex: &str, out: &mut [u8]) -> Result<(), ParseError> {
     Ok(())
 }
 
+/// Serialize a group element into ZCash format.
+pub fn write_g<P: SWModelParameters>(point: &GroupAffine<P>) -> String
+where
+    <P as ModelParameters>::BaseField: ToBasePrimeFieldIterator,
+{
+    type FieldOf<P> = <P as ModelParameters>::BaseField;
+    let ext_degree: usize = FieldOf::<P>::extension_degree().try_into().unwrap();
+    let element_size = <<FieldOf<P> as Field>::BasePrimeField as PrimeField>::BigInt::NUM_LIMBS * 8;
+    let mut buf = vec![0u8; element_size * ext_degree];
+    buf.chunks_exact_mut(element_size)
+        .zip(point.x.base_field_iterator().rev())
+        .for_each(|(chunk, element)| {
+            let repr = element.into_repr();
+            let mut writer = &mut chunk[..];
+            repr.write_le(&mut writer).unwrap();
+            chunk.reverse();
+        });
+    buf[0] |= 0x80; // compressed
+    if point.infinity {
+        buf[0] |= 0x40;
```

Comment:
The spec requires all other bits to be zero. I don't know if Arkworks guarantees this.